src/NHibernate/Hql/Ast/ANTLR/QueryTranslatorImpl.cs

```diff
@@ -283,9 +284,29 @@ public bool ContainsCollectionFetches
 		{
 			get
 			{
-				ErrorIfDML();
-				IList<IASTNode> collectionFetches = ((QueryNode)_sqlAst).FromClause.GetCollectionFetches();
-				return collectionFetches != null && collectionFetches.Count > 0;
+				return CollectionFetches.Count > 0;
+			}
+		}
+
+		public bool CanAddFetchedCollectionToCache
```

Comment:
This probably needs to be moved to a loader, however, I'm not entirely sure.

Reply:
And shouldn't it be `CanAddFetchedCollectionsToCache`? (Collection is switched to plural.)

If I understand well this PR, contrary to #460, it would disable collection caching for the whole query, not just for the one having restrictions.

> This probably needs to be moved to a loader, however, I'm not entirely sure.

I am not sure either. It may require to expose some internal state of the translator for the loader to analyse it, and then, better keep it in the translator.